lib/streamlit/elements/arrow_vega_lite.py

```diff
@@ -214,15 +214,14 @@ def vega_lite_chart(
                 raise StreamlitAPIException(
                     "In order to make VegaLite work, one needs to have a selection enabled through add_params. Please check out this documentation to add some: https://altair-viz.github.io/user_guide/interactions.html#selections-capturing-chart-interactions"
                 )
+            has_selection_object = False
             for param in spec["params"]:
-                if (
-                    "name" not in param
-                    or "select" not in param
-                    or "type" not in param["select"]
-                ):
-                    raise StreamlitAPIException(
-                        "In order to make VegaLite work, one needs to have a selection enabled through add_params. Please check out this documentation to add some: https://altair-viz.github.io/user_guide/interactions.html#selections-capturing-chart-interactions"
-                    )
+                if "name" in param and "select" in param and "type" in param["select"]:
+                    has_selection_object = True
+            if not has_selection_object:
+                raise StreamlitAPIException(
+                    "In order to make VegaLite work, one needs to have a selection enabled through add_params. Please check out this documentation to add some: https://altair-viz.github.io/user_guide/interactions.html#selections-capturing-chart-interactions"
```

Comment:
This is also the same text as the one in line 215. Can we put that into a constant? Btw, the linked link only explains it for altair, but since this is vega lite, is there also a link showing how to do that in vega lite syntax?

Reply:
Good catch. I'll add something.